src/searchReplace.tsx

```diff
@@ -2,17 +2,27 @@ import React from 'react';
 import { Debouncer } from '@lumino/polling';
 import { requestAPI } from './handler';
 import { VDomModel, VDomRenderer } from '@jupyterlab/apputils';
-import { Search } from '@jupyter-notebook/react-components';
+import { Search, Progress } from '@jupyter-notebook/react-components';
 
 export class SearchReplaceModel extends VDomModel {
   constructor() {
     super();
+    this._isLoading = false;
     this._searchString = '';
     this._debouncedStartSearch = new Debouncer(() => {
       this.getSearchString(this._searchString);
     });
   }
 
+  get isLoading(): boolean {
+    return this._isLoading;
+  }
+
+  set isLoading(v: boolean) {
+    this._isLoading = v;
+    this.stateChanged.emit();
```

Comment:
Let's emit the signal only if `v !== this._isLoading`.

Reply:
done